pyqtorch/modules/parametric.py

```diff
@@ -22,6 +22,51 @@ def __init__(self, gate: str, qubits: ArrayLike, n_qubits: int):
         self.register_buffer("imat", OPERATIONS_DICT["I"])
         self.register_buffer("paulimat", OPERATIONS_DICT[gate])
 
+    def __mul__(self, other: RotationGate | QuantumCircuit) -> QuantumCircuit:
```

Comment:
This should also work for the primitive `X`/`Y`/etc. gates

Reply:
Having an `AbstractGate` class as a parent to all gate classes would help here.